File: java/conway/src/main/java/conway/blocks/Block.java
package conway.blocks;

import java.io.Serializable;


public class Block implements Serializable {

    // Serializer version code
    private static final long serialVersionUID = 2L;

    // Matrix
    private int[][] matrix;


    // Random
    public Block() {
        this.matrix = new int[Conway.B_SIZE][Conway.B_SIZE];

        for (int i = 0; i < Conway.B_SIZE; ++i) {
            for (int j = 0; j < Conway.B_SIZE; ++j) {
                this.matrix[i][j] = (int) Math.floor(Math.random() * 2);
            }
        }
    }

    // Hard copy
    public Block(Block ref) {
        this.matrix = new int[Conway.B_SIZE][Conway.B_SIZE];

        for (int i = 0; i < Conway.B_SIZE; ++i) {
            for (int j = 0; j < Conway.B_SIZE; ++j) {
                this.matrix[i][j] = ref.matrix[i][j];
            }
        }
    }

    // Getters and setters
    public void set(int i, int j, int val) {
        this.matrix[i][j] = val;
    }

    public int get(int i, int j) {
        return this.matrix[i][j];
    }

}

File: python/examples/conway/src/block.py
class Block (object):
    def __init__(self, blockSize = None, ref = None, rand = 0):
        from random import randint
        if(ref == None):
            if rand == 1:
                self._blockSize = blockSize
                self._matrix = []
                for i in range(0, self._blockSize):
                    a = []
                    for j in range(0, self._blockSize):
                        a.append(randint(0, 1))
                    self._matrix.append(a)
            else:
                self._blockSize = blockSize
                self._matrix = []
                for i in range(0, self._blockSize):
                    a = []
                    for j in range(0, self._blockSize):
                        a.append(0)
                    self._matrix.append(a)
        else:
            self._blockSize = ref._blockSize
            self._matrix = []
            for i in range(0, self._blockSize):
                a = []
                for j in range(0, self._blockSize):
                    a.append(ref._matrix[i][j])
                self._matrix.append(a)

    def set (self, i, j, val):
        self._matrix[i][j] = val

    def get (self, i, j):
        return self._matrix[i][j]